docs/src/references/synchronize-daemons.md

```diff
@@ -0,0 +1,116 @@
+
+## Proposal 1: Use the `health check` mechanism
+
+The mapper uses the health-check mechanism to detect the health of another daemon and perform all the cloud-specific
+ actions for that daemon like updating the supported operations etc.
+In this, the `daemon` sends the health status `up` message once it's completely up. Then the `mapper` picks up this message and creates the required operation files and also sends the supported operation to the cloud.
+Also, now the mapper can start interacting with the daemon.
+
+Example 1,
+- When tedge-mapper-c8y is started, it will subscribe to `tedge/health/#` topic
+- When tedge-agent is started, it will publish health status as `up` on to `tedge/health/tedge-agent` topic
+- Now the mapper picks up this message and creates the supported operations file (c8y_SoftwareUpdate, etc) in `/etc/tedge/operations/c8y` with their content. 
+- Mapper sends the updated supported operations list to the c8y cloud.
+- Also, now that the tedge-agent is up and running, the tedge-mapper-c8y will publish the request to get the
+ software list on `tedge/commands/req/software/list`.
+- Now the tedge-agent processes this request and sends the software list to the mapper, once the mapper gets the response it will forward it to the c8y cloud.
+
+Example 2,
+- When tedge-mapper-c8y is started, it will subscribe to `tedge/health/#` topic
+- When c8y-log-plugin is started, it will publish health status as `up` on to `tedge/health/c8y-log-plugin` topic
+- Now the mapper picks up this message and creates the c8y_LogfileRequest operation file in `/etc/tedge/operations/c8y`.
+- Also the mapper creates the `c8y-log-plugin.toml` file, which contains the `supported log types` in `/etc/tedge/c8y`.
+- Mapper sends the updated supported operations list to the c8y cloud. Also, it sends the supported `log types`(118,software-management) to the c8y cloud.
+
+Pros: 
+-   This will remove the file system dependency when the daemons want to create the supported operation files on the mapper file system.
+    For example, tedge-agent wants to create the supported operations files in /etc/tedge/operations/c8y directory.
+
+Cons: 
+-   This leads to hard dependency on the list of daemons in the mapper. For example, the tedge-mapper-c8y has to subscribe to tedge/health/tedge-agent
+topic to receive the health status from the tedge-agent. Based on this the mapper has to create a supported operation file in /etc/tedge/operations/c8y.
+-   This will be a hard-coded dependency of the name of the plugin and can’t be updated on the fly if someone wants to change the name of the daemon
+    or wants to add a new daemon.
+
+## Proposal 2:  Introduce an `init message` mechanism
+
+Make all the tedge components publish a `retained init` message describing their role (software management, config management, etc)
+ and other details specific to that role.
+
+The proposed topic will be `tedge/<device-id>/init/<cloud>`.
```

Comment:
Although I understand the need for cloud specific init endpoints(so that devs can develop cloud specific plugins as well), we should also provide cloud agnostic init endpoints like just tedge/init to which cloud agnostic plugins can declare their capabilities in a cloud agnostic manner. If we only support cloud specific init endpoints, we'll be forcing all the plugins also to be cloud aware, which is against our cloud-agnostic principle.

Reply:
Yes, I changed the spec. Now the `cloud-specific` mapper will subscribe to both the `tedge/<device-id>/init` as well as `tedge/<device-id>/init/<cloud>` both topics. The plugins have the freedom to send the init message either to the `cloud-specific` or to the `generic` init topic.